### ISSUE TITLE
Can't open from reaper

### ISSUE THREAD

**Issue by pepperpepperpepper** [amsynth/amsynth#143]
### Expected behavior
open an amsynth instance from reaper as a VSTi virtual instrument 
### Actual behavior
reaper and amsynth both crash
### Information

  - Which version of amsynth are you using?
the latest version from github (1.9)
  - Which build of amsynth are you using? (stand-alone application, LV2, DSSI, VST)
VST
  - If using amsynth as a plugin, which host application are you using?
reaper
  - Which operating system & version are you using?
Linux
### Here's the error log from Reaper
(reaper:14213): GLib-GObject-WARNING **: 17:36:28.523: cannot register existing type 'GdkWindow'

(reaper:14213): GLib-GObject-CRITICAL **: 17:36:28.523: g_param_spec_object: assertion 'g_type_is_a (object_type, G_TYPE_OBJECT)' failed

(reaper:14213): GLib-GObject-CRITICAL **: 17:36:28.523: validate_pspec_to_install: assertion 'G_IS_PARAM_SPEC (pspec)' failed

(reaper:14213): GLib-GObject-WARNING **: 17:36:28.594: cannot register existing type 'GdkScreen'

(reaper:14213): GLib-CRITICAL **: 17:36:28.594: g_once_init_leave: assertion 'result != 0' failed

(reaper:14213): Gdk-CRITICAL **: 17:36:28.594: IA__gdk_screen_get_default_colormap: assertion 'GDK_IS_SCREEN (screen)' failed

(reaper:14213): Gdk-CRITICAL **: 17:36:28.594: IA__gdk_colormap_get_visual: assertion 'GDK_IS_COLORMAP (colormap)' failed

  
It seems like there's some sort of conflict happening with GDK. Any ideas about this would be much appreciated!!! What do u think?

**Comment by nickdowell** [comment]
Unfortunately amsynth is not current compatible with REAPER 😞 

Rebuilding Reaper's `libSwell.so` with GDK2 instead of GDK3 (as documented in its `readme-linux.txt`) slightly improves the situation, but there are still major issues that make it unusable in practice.

See earlier comments - https://github.com/amsynth/amsynth/issues/140#issuecomment-497572218

**Comment by nickdowell** [comment]
I've updated the VST version of amsynth to disable the GUI under REAPER - this allows the plugin to be loaded and used with REAPER's generic controls.

**Comment by pigwink** [comment]
I'm using the 1.9 vst and it still crashes Reaper before it even loads.

**Comment by nickdowell** [comment]
@pigwink are you able to build amsynth from code to try the REAPER workaround? Version 1.9 does not contain it...

**Comment by pigwink** [comment]
Removed 1.9 and built 1.8. VST still crashes Reaper.

**Comment by nickdowell** [comment]
@pigwink 1.8 doesn't contain the workaround either, please try using the latest code from git;

```
git clone https://github.com/amsynth/amsynth.git
cd amsynth
./autogen.sh
./configure
make
sudo make install
```

**Comment by pigwink** [comment]
Works now with generic GUI. But now it's dropping the notes it receives from my keyboard. I can see Reaper is receiving midi from keyboard, but some keys are not sounding. Weird, never seen before.

**Comment by nickdowell** [comment]
@pigwink thanks for the update; I need to investigate the the dropped notes issue.

**Comment by falkTX** [comment]
Mixing gtk3 and gtk2 in the same process space is really problematic.
It might not crash at first, but we cannot predict what nasty things are happening in the background. Some of the host gdk/gtk3 functions will point to gdk/gtk2 ones after loading amsynth.

This issue is also present on zrythm, which uses gtk3 for its GUI.

The only proper solution for this is to stop using general graphical toolkits like Gtk or Qt for plugin UIs (unless you statically build and hide their symbols, but easier said than done).
Note that Ardour 6 now removed support for any non-X11 plugin UI (so that means no embeding Gtk or Qt based UIs).
There are just far too many problems when using these toolkits in a plugin, as plugins should really be as self-contained as possible. Messing with global state is very nasty and leads to issues like this.

**Comment by nickdowell** [comment]
@pigwink the dropped note issue is now fixed on the `develop` branch

**Comment by pigwink** [comment]
Great! I'll check it out soon.

**Comment by pigwink** [comment]
When running ./configure i get: ./configure: line 11745: syntax error near unexpected token `noext,'
./configure: line 11745: `AX_CXX_COMPILE_STDCXX_11(noext, mandatory)'

**Comment by pigwink** [comment]
no makefile found

**Comment by nickdowell** [comment]
@pigwink looks like the "autoconf archive" package needs to be installed on your system...

On Debian / Ubuntu it's `apt install autoconf-archive`

**Comment by pigwink** [comment]
That made no difference. Same errors and no makefile.

**Comment by pigwink** [comment]
got it working now!

**Comment by pigwink** [comment]
After building, can't find the .so -file?

**Comment by nickdowell** [comment]
Ah, automake puts in a directory called .libs

**Comment by pigwink** [comment]
i can see .libs but it's empty

**Comment by nickdowell** [comment]
Odd; after running `make` I have `.libs/amsynth_vst.so`

The last few lines of make output are
```
  CXXLD    amsynth_dssi_gtk
  CXXLD    amsynth
  CXXLD    amsynth_dssi.la
  CXXLD    amsynth_vst.la
  CXXLD    amsynth_lv2_gtk.la
  CXXLD    amsynth_lv2.la
```

**Comment by pigwink** [comment]
getting this after 'make':

make  all-recursive
make[1]: Entering directory '/home/pigwink/amsynth'
Making all in po
make[2]: Entering directory '/home/pigwink/amsynth/po'
  MSGFMT de.gmo
  MSGFMT fr.gmo
make[2]: Leaving directory '/home/pigwink/amsynth/po'
make[2]: Entering directory '/home/pigwink/amsynth'
  CXX      src/amsynth-Configuration.o
  CXX      src/amsynth-filesystem.o
  CXX      src/amsynth-MidiController.o
  CXX      src/amsynth-Parameter.o
  CXX      src/amsynth-Preset.o
  CXX      src/amsynth-PresetController.o
  CXX      src/Effects/amsynth-Distortion.o
  CXX      src/Effects/amsynth-SoftLimiter.o
  CXX      src/freeverb/amsynth-allpass.o
  CXX      src/freeverb/amsynth-comb.o
  CXX      src/freeverb/amsynth-revmodel.o
  CXX      src/amsynth-Synthesizer.o
  CXX      src/amsynth-TuningMap.o
  CXX      src/amsynth-VoiceAllocationUnit.o
  CXX      src/VoiceBoard/amsynth-ADSR.o
  CXX      src/VoiceBoard/amsynth-LowPassFilter.o
  CXX      src/VoiceBoard/amsynth-Oscillator.o
  CXX      src/VoiceBoard/amsynth-VoiceBoard.o
  CXX      src/amsynth-AudioOutput.o
  CXX      src/drivers/amsynth-ALSAAudioDriver.o
  CXX      src/drivers/amsynth-ALSAMidiDriver.o
  CXX      src/drivers/amsynth-ALSAmmapAudioDriver.o
  CXX      src/drivers/amsynth-OSSAudioDriver.o
  CXX      src/drivers/amsynth-OSSMidiDriver.o
  CXX      src/amsynth-JackOutput.o
  CC       src/amsynth-lash.o
  CXX      src/amsynth-main.o
  CC       src/GUI/amsynth-bitmap_button.o
src/GUI/bitmap_button.c:38:1: warning: no semicolon at end of struct or union
   38 | } bitmap_button;
      | ^
  CC       src/GUI/amsynth-bitmap_knob.o
  CC       src/GUI/amsynth-bitmap_popup.o
  CXX      src/GUI/amsynth-editor_menus.o
  CC       src/GUI/amsynth-editor_pane.o
  CXX      src/GUI/amsynth-gui_main.o
  CXX      src/GUI/amsynth-ConfigDialog.o
  CXX      src/GUI/amsynth-MainMenu.o
  CXX      src/GUI/amsynth-MainWindow.o
  CXX      src/GUI/amsynth-MIDILearnDialog.o
  CXX      src/GUI/amsynth-PresetControllerView.o
  CXXLD    amsynth
/usr/bin/ld: src/amsynth-AudioOutput.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:1926: amsynth] Error 1
make[2]: Leaving directory '/home/pigwink/amsynth'
make[1]: *** [Makefile:3699: all-recursive] Error 1
make[1]: Leaving directory '/home/pigwink/amsynth'
make: *** [Makefile:1479: all] Error 2

**Comment by nickdowell** [comment]
Thanks for the info, that’s an error I don’t see on my system. Which flavour of Linux are you running?

**Comment by pigwink** [comment]
Ubuntu Studio

**Comment by pigwink** [comment]
I also have the kxstudio repos installed

**Comment by nickdowell** [comment]
It looks like you may not have the latest version of the code, since amsynth stopped using the pthread API in release 1.11.0

Could you check by running `git checkout develop && git pull --rebase`?

**Comment by pigwink** [comment]
fatal: not a git repository (or any of the parent directories): .git

**Comment by pigwink** [comment]
sorry.

Already on 'develop'
Your branch is up to date with 'origin/develop'.
Already up to date.
Current branch develop is up to date.

**Comment by nickdowell** [comment]
I can reproduce the build failure on Ubuntu Studio 20.10 / GCC 10.2.0

**Comment by nickdowell** [comment]
Ubuntu 20.10 is still in beta and is not yet available on Travis, GitHub, or Launchpad CI, which is why this issue hasn't been surfaced before...

**Comment by nickdowell** [comment]
Hi @pigwink the linker error is fixed now 😅

**Comment by pigwink** [comment]
It's working in reaper now, and no dropped notes! (and no GUI)
I'm on Ubuntu 20.04.1 LTS (Focal Fossa).
Is there any way to load the presets without the GUI?

**Comment by nickdowell** [comment]
The VST version doesn't currently have a way to load presets... let me see what can be done

**Comment by nickdowell** [comment]
I've just pushed a change to `develop` which allows selecting presets through the generic VST UI :tada:

There are a LOT of presets, which means the UI isn't ideal... but it's better than nothing :grin:

**Comment by pigwink** [comment]
Great! I'll check it out after the weekend!

**Comment by pepperpepperpepper** [comment]
Thanks so much for this amazing synth. Are tuning settings included in the
presets?

**Comment by nickdowell** [comment]
The presets do not contain the alternate tuning settings

**Comment by nickdowell** [comment]
The fixes are in the 1.12.0 release :tada:
